### DISCUSSION TITLE
Add fragment selection for element-type links

### DISCUSSION THREAD

**Discussion by engram-design** [verbb/hyper#38] [Ideas]
There are times where you might want to link to a fragment on a page - for example linking to an ID on a page.

You might have: `<h2 id="my-heading">Example Heading</h2>` and you'd want to provide a nice means to link to that. This would only be possible for element-type links, and not 100% sure on the UI for this, combining with element select fields. But you would select the element you want, an Ajax request to query the front-end template for that element, parse just items with an ID attribute, and list those for picking.

**Comment by MangoMarcus** [comment]
+1 for fragment support. The typed link plugin supported this with just a free text field you could enable in the field settings

<img width="816" alt="Screenshot 2024-09-30 at 13 29 06" src="https://github.com/user-attachments/assets/46ba2a51-0e7c-4327-9d0b-4185a6b07488">

Not the most intuitive solution for content editors but with some field instructions for clarification I think it does the job

**Reply by engram-design** [reply]
Certainly something you could do on your own with Hyper's URL Suffix field. You could even put it on the first tab to visually make it look like Typed Link.

**Comment by brettburwell** [comment]
Support for something like this would be amazing. 

I frequently need to provide functionality to link directly to a specific Matrix block on a page (where the page has corresponding anchor links for individual blocks). I'm currently heading down a path similar to what @MangoMarcus is doing above, and then will append the anchor to the URL manually in my templates. 

Being able to select from Matrix blocks included on a specific page (and somehow have that auto-converted into a URL with anchor) would be, for me, the dream scenario. I'd imagine providing full support for something like this might be messy and difficult. 

A start might be adding a special "Anchor Link" link type that adds a plain text field for the anchor (similar to above), but with the bonus of Hyper handling adding that anchor to the URL output for you. Even that bit of automation would be nifty and help keep the template / partial code a little cleaner.

**Reply by engram-design** [reply]
FYI, you can use Hyper's URL Suffix field to save you having to manually construct the URL.

The "advanced" option will be an interesting one to implement. As I mentioned, not 100% sure how to go about it just yet, but I think it'd be pretty neat to provide a client-friendly way to target "sections" on a page. Definitely something we want to at least try and tackle.

You could actually implement that yourself, creating a new link type from "Custom", name it Anchor, and then provide the values as `#my-anchor`.

**Reply by brettburwell** [reply]
I feel silly saying that I didn't even realize Hyper had the URL Suffix field. That worked like a charm. Thanks @engram-design! 

And if you ever figure out a reasonable way to tackle the "advanced" option, that'd be dandy, but this is a totally workable solution for now. Have a good weekend.
